src/launchpad/service.py

```diff
@@ -167,10 +187,33 @@ def handle_kafka_message(self, payload: PreprodArtifactEvents) -> None:
     def process_artifact(self, artifact_id: str, project_id: str, organization_id: str) -> None:
         """
         Download artifact and perform size analysis.
+
+        Uses subprocess isolation to ensure all memory is freed after processing,
+        preventing memory leaks from C extensions like LIEF and pillow_heif.
         """
         if not self._service_config:
             raise RuntimeError("Service not properly initialized. Call setup() first.")
 
+        process = multiprocessing.get_context("spawn").Process(
```

Comment:
Saw this in passing and I'm super in favour of this! I think it will solve a lot of problems, and also be good for Android. Rather than using `multiprocessing` how do you feel about calling `subprocess.check_call("launchpad", "size", path_to_app, "-o", analysis_result)` directly and still doing the upload from the main process?

My reasoning being:
- We have all kinds of weird threads and network connections (kafka, arroyo, datadog, sentry client, sentry sdk, etc) which are the kinds of things that behave badly with fork/spawn
- If we reuse the local size command line that's one less code path to worry about
- We can keep central control over the connection to the monolith which is nice for logging / retries / queuing / kill switches etc.

Reply:
Great suggestions.

For context Nico and I were speaking about this offline and I asked him to postpone merging this for now. I was thinking it may be easier to debug the memory issues we've been seeing while everything is still single-process, and then try to merge this in afterwards. 

I'm also not clear how this would fit in with one consumer potentially processing multiple messages in parallel, which is similar to how the iOS infra worked at Emerge. I see that Arroyo actually supports something called `RunTaskWithMultiprocessing` which might handle this automatically for us? Currently we are using `RunTaskInThreads` so I'm not sure if we've experimented with that before, it does seem like something the framework could handle for us.